File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/Banshee.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import androidx.recyclerview.widget.ItemTouchHelper;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class Banshee extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 205;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 153;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 1400;
        this.baseConstitution = ItemTouchHelper.Callback.DEFAULT_SWIPE_ANIMATION_DURATION;
        this.baseIntelligence = 5;
        this.baseDexterity = 85;
        this.baseDefense = 35;
        this.baseMagicDefense = 35;
        this.team = 2;
        this.imageId = R.drawable.unit_banshee;
        this.idName = R.string.enemy_banshee_name;
        this.idDescription = R.string.enemy_banshee_description;
        this.threat = 3;
        this.passiveSkill = Skills.PASSIVE_TERRITORIAL;
        this.activeSkill = Skills.ACTIVE_NONE;
        this.rarity = 1;
        this.expGiven = ItemTouchHelper.Callback.DEFAULT_SWIPE_ANIMATION_DURATION;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("BansheeScale", 1), 600);
        linkedHashMap.put(ItemWrapper.getInstance("BansheeClaw", 1), 18);
        linkedHashMap.put(ItemWrapper.getInstance("BansheeHorn", 1), 2);
        linkedHashMap.put(ItemWrapper.getInstance("ReptileEgg", 1), 1);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/Banshee.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class Banshee : Enemy() {
    override fun getMaxDamage(): Int = 205
    override fun getMinDamage(): Int = 153
    override fun isMagic(): Boolean = false
    override fun isRanged(): Boolean = false

    override fun configureStatistics() {
        baseMaxHp = 1400
        baseConstitution = 250
        baseIntelligence = 5
        baseDexterity = 85
        baseDefense = 35
        baseMagicDefense = 35
        team = 2
        imageId = R.drawable.unit_banshee
        idName = R.string.enemy_banshee_name
        idDescription = R.string.enemy_banshee_description
        threat = 3
        passiveSkill = Skills.PASSIVE_TERRITORIAL
        activeSkill = Skills.ACTIVE_NONE
        rarity = 1
        expGiven = 250
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("BansheeScale", 1), 600)
        linkedHashMap.put(ItemWrapper.getInstance("BansheeClaw", 1), 18)
        linkedHashMap.put(ItemWrapper.getInstance("BansheeHorn", 1), 2)
        linkedHashMap.put(ItemWrapper.getInstance("ReptileEgg", 1), 1)
        return linkedHashMap
    }
}
